src/hooks/item.ts

```diff
@@ -63,21 +63,31 @@ export default (
       : undefined;
 
   return {
-    useOwnItems: (options?: { getUpdates?: boolean }) => {
+    useOwnItems: (
+      { page, all, name, limit }: OwnItemsQuery,
+      options?: { getUpdates?: boolean },
+    ) => {
       const queryClient = useQueryClient();
       const getUpdates = options?.getUpdates ?? enableWebsocket;
 
       const { data: currentMember } = useCurrentMember();
       itemWsHooks?.useOwnItemsUpdates(getUpdates ? currentMember?.id : null);
 
       return useQuery({
-        queryKey: OWN_ITEMS_KEY,
+        queryKey: buildOwnItemsKey({ page, name, all, limit }),
         queryFn: (): Promise<List<ItemRecord>> =>
```

Comment:
This type is not correct anymore. Most probably we want a type such as:
```
ImmutableCast<Paginated<Item>>

type Paginated<T> = {
  totalCount: number, 
  data: T[]
}
```
I think this should be in sdk. Also because you will need to have an immutable type (I can explain to you on Tuesday!)